### PR TITLE
Setup integration testing for Android Plugin

### PR DESCRIPTION
**Pull request by tresat**
- Set up an integration test suite that can apply the ecosystem plugin.
- Create an internal test utils lib containing many test file utils from gradle/gradle.
- Demo tests for applying the plugin with and without a convention.

### CODE REVIEW

File: unified-prototype/unified-plugin/internal-testing-utils/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+@file:Suppress("UnstableApiUsage")
+
+plugins {
+    groovy
+    `java-test-fixtures`
+    `java-library`
+}
+
+description = "Implements the declarative Android DSL prototype"
```
Comment (oleg-nenashev) [inline unified-prototype/unified-plugin/internal-testing-utils/build.gradle.kts:9]: Copy-paste?

**Review by oleg-nenashev — COMMENTED** [review]
Should it be in another directory instead of `unified-prototype`? Maybe time to start assembling the toolchain/test tools directory.

P.S: I have not reviewed the main code. We copy over quite a lot of content, aren't we?

File: unified-prototype/unified-plugin/internal-testing-utils/build.gradle.kts (hunk 1 above)
Comment (tresat) [reply unified-prototype/unified-plugin/internal-testing-utils/build.gradle.kts:9]: Yep.

**Comment by tresat** [comment]
> Should it be in another directory instead of `unified-prototype`? Maybe time to start assembling the toolchain/test tools directory.

Happy to move it whereever.  Hopefully it should be self-contained and portable.  Maybe the JVM team can review the current structure of this repo soon and make some recommendations.

**Pull request merged**